src/_cffi_src/openssl/src/osrandom_engine.c

```diff
@@ -90,9 +94,54 @@ static struct {
     ino_t st_ino;
 } urandom_cache = { -1 };
 
+static int set_cloexec(int fd) {
+    int flags = fcntl(fd, F_GETFD);
+    if (flags == -1) {
+        return -1;
+    }
+    if (fcntl(fd, F_SETFD, flags | FD_CLOEXEC) == -1) {
+        return -1;
+    }
+    return 0;
+}
+
+static void close_fd(int fd) {
+    int n;
+    do {
+        n = close(fd);
+    } while (n < 0 && errno == EINTR);
```

Comment:
This is not thread-safe (maybe not a problem in Python?) and is unnecessary on Linux. See http://ewontfix.com/4/